### PR TITLE
cmd/snap: packaging: move snap binary to /usr/lib/snapd/snap, fix invocation of internal tools

### PR DESCRIPTION
**Pull request by bboozzoo**
On Core systems with the snapd snap, /usr/bin/snap is a symlink pointing to
/snap/snapd/current/usr/bin/snap. However, while the snapd snap is being
refreshed, the current symlink becomes invalid for a short while. Should a
reboot occur while this happens, the existing snap services will be unable to
start as /usr/bin/snap points to an invalid location.

To address this, we move the snap binary to /usr/lib/snapd/snap, as /usr/lib/snapd gets exported as the usr-lib-snapd.mount. Then the core18/20 snap has /usr/bin/snap which is a wrapper script that picks /snap/snap/current/usr/lib/snap or /usr/lib/snapd/snap.

### CODE REVIEW

**Comment by pedronis** [comment]
as discussed with @bboozzoo there is still more to do related to invoking snap-confine and having the right profile available

File: cmd/snap/cmd_run.go [hunk 1]
```diff
@@ -517,13 +518,22 @@ func snapdHelperPath(toolName string) (string, error) {
 	// The logic below only works if the last two path components
 	// are /usr/bin
 	// FIXME: use a snap warning?
-	if !strings.HasSuffix(exe, "/usr/bin/"+filepath.Base(exe)) {
+	baseName := filepath.Base(exe)
+	if !strings.HasSuffix(exe, "/usr/bin/"+baseName) && !strings.HasSuffix(exe, filepath.Join(dirs.CoreLibExecDir, baseName)) {
```
Comment (anonymouse64) [inline cmd/snap/cmd_run.go:522]: this change feels like it needs a unit test, no?

**Review by anonymouse64 — COMMENTED** [review]
lgtm mostly, but I would feel better about the changes in cmd_run.go if there were unit tests too, I'm a little unsure if the behavior is quite correct or not without seeing any test additions

**Comment by pedronis** [comment]
we likely don't need this

**Comment by bboozzoo** [comment]
Closing in favor or #10067 (for not at least)

**Pull request closed without merging**